go/consensus/tendermint/abci/system.go

```diff
@@ -0,0 +1,122 @@
+package abci
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/cometbft/cometbft/abci/types"
+
+	"github.com/oasisprotocol/oasis-core/go/common/cbor"
+	consensus "github.com/oasisprotocol/oasis-core/go/consensus/api"
+	"github.com/oasisprotocol/oasis-core/go/consensus/api/transaction"
+	"github.com/oasisprotocol/oasis-core/go/consensus/tendermint/api"
+	tmcrypto "github.com/oasisprotocol/oasis-core/go/consensus/tendermint/crypto"
+)
+
+// prepareSystemTxs prepares a list of system transactions to be included in a proposed block in
+// case where the node is currently the block proposer.
+func (mux *abciMux) prepareSystemTxs() [][]byte {
+	var systemTxs [][]byte
+
+	// Append block metadata as a system transaction.
```

Comment:
We could for loop over `SystemMethods`, do a `switch`  and put this code under `case consensus.MethodMeta`.